compose.yaml

```diff
@@ -0,0 +1,42 @@
+version: '3.8'
+
+services:
+  frontend:
+    build:
+      context: frontend
+      dockerfile: Dockerfile
+    networks:
+      - opendevin
+    ports:
+      - "127.0.0.1:3001:3001"
+    container_name: devin-frontend
+    depends_on:
+      - backend
+    volumes:
+      - ./config.toml:/usr/src/app/config.toml
+
+    command: ["npm", "start", "--", "--host", "0.0.0.0"]
+
+  backend:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    networks:
+      - opendevin
+    ports:
+      - "127.0.0.1:3000:3000"
```

Comment:
1. Maybe consider leaving "127.0.0.1:" off so that it will come up on 0.0.0.0 so that this could be deployed to a shared server?
2. Maybe consider using a `.env` file so that it is more clear these are are configurable?
```yaml
    frontend:
        <...snip...>
        ports:
            - "${PUBLIC_PORT_FRONTEND}:3001"
        env_file:
            - ./.env
        environment:
            - VITE_URL=http://${PUBLIC_HOST_BACKEND}:${PUBLIC_PORT_BACKEND}
            - VITE_WS_URL=ws://${PUBLIC_HOST_BACKEND}:${PUBLIC_PORT_BACKEND}/ws
        <...snip...>

   backend:
        <...snip...>
        ports:
            - "${PUBLIC_PORT_BACKEND}:3000"
        env_file:
            - ./.env
        <...snip...>
```
### `.env`
```bash
# Maybe a host name other than localhost
PUBLIC_HOST_BACKEND=127.0.0.1 
# Maybe 3330 if another app is running on docker public port 3000
PUBLIC_PORT_BACKEND=3000 

# maybe a host name other than localhost
PUBLIC_HOST_FRONTEND=127.0.0.1  
# Maybe 3331 if another app is running on docker public port 3001
PUBLIC_PORT_FRONTEND=3001 
```

Reply:
The `127.0.0.1:3000:3000` is required for users who are running in OSX or Windows and are using a docker machine. 

I can look into alternatives, however it is required for OSX users. 

One alternative is to do host networking, however from my testing I ran into some problems with that.

Reply:
Ah, I think you are referring to CORS errors, because only localhost:3001 is accepted:

https://github.com/OpenDevin/OpenDevin/blob/main/opendevin/server/listen.py#L12C8-L12C9

```python
app.add_middleware(
    CORSMiddleware,
    allow_origins=["http://localhost:3001"],
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)
```

Reply:
I was able to get `3000:3000` working without the `127.0.0.1:3000:3000`. Thanks for pointing that out so I could optimize it.